Damage.cs

```diff
@@ -17,49 +16,54 @@ public EDamageTypes DamageType
                 return damageType;
             }
         }
-        #endregion
 
-        #region TargetAttribute
         public struct Target
         {
             public float Armor;
 
             public ERA ERAData;
+
+            public float Health;
         }
 
         public struct ERA
         {
-            public float currentValue;
+            public float CurrentValue;
 
             public float KEFractionMultiplier;
 
-            public float HEATFractionMultiplier;
+            public float HeatFractionMultiplier;
         }
-        #endregion
 
-        public Damage(EDamageTypes dt)
+        protected Target _target;
+
+        public Damage(EDamageTypes dt, Target t)
         {
             // In base class constructor, pass in the type of damage
             damageType = dt;
+            // And the target
+            _target = t;
         }
 
         /// <Summary>
-        /// Use this method to deal damage
-        /// This method only calculates HP damage, deal with armour degradation in separate method
+        /// Use this method to deal damage.
+        /// This method returns a Target struct
+        /// containing updated values of Health, Armor and ERA stats.
         /// </Summary>
-        virtual public float DealDamage()
+        virtual public Target CalculateDamage()
         {
             // Override this function to specify damage algorithm
-            return 0.0;
+            Target finalState;
+            return finalState;
         }
     }
 
     public enum EDamageTypes
```

Comment:
We want to avoid anything resembling hungarian notation. Please stick to the convention of `DamageTypes`. 

PS: You didn't do anything wrong but enums should not be named as a singular noun but plural (like you did!).